### PR TITLE
Add a self-aware bot if they're waiting in a matchme room

### PR DESCRIPTION
**Pull request by vpzomtrrfrt**
This might be a thing we want to do

### CODE REVIEW

File: stupidclient.js [hunk 1]
```diff
@@ -1,97 +1,114 @@
...
+	};
+	s.onclose = function() {
+		if(opts.log) console.log("Lost connection");
+		if(require.main === module) {
+			process.exit();
+		}
+	};
+	setInterval(function(){
+		if(ai && ai.active) {
+			if(opts.log) console.log(ai.mynodes);
```
Comment (vpzomtrrfrt) [inline stupidclient.js]: Codacy Issue found: [Expected { after 'if' condition.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork-server/file/2531799817/issues/source?bid=3248399&fileBranchId=3284044#l97)

File: stupidclient.js [hunk 2]
```diff
@@ -1,97 +1,114 @@
...
-};
-s.onclose = function() {
-	console.log("Lost connection");
-	process.exit();
+		else if(ai && d.data == "gamestart") {
+			ai.active = true;
+		}
+	};
+	s.onclose = function() {
+		if(opts.log) console.log("Lost connection");
```
Comment (vpzomtrrfrt) [inline stupidclient.js]: Codacy Issue found: [Expected { after 'if' condition.](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork-server/file/2531799817/issues/source?bid=3248399&fileBranchId=3284044#l90)

File: stupidclient.js (hunk 1 above)
Comment (riftEmber) [inline stupidclient.js]: Is it automatically adding these notes?

**Comment by riftEmber** [comment]
So the bot would run on the server?

File: stupidclient.js (hunk 1 above)
Comment (vpzomtrrfrt) [inline stupidclient.js]: I guess.  Wow.

**Comment by vpzomtrrfrt** [comment]
Kinda, the server spawns stupidclients.

File: stupidclient.js (hunk 1 above)
Comment (riftEmber) [inline stupidclient.js]: I thought you would have set it up lol

Comment (vpzomtrrfrt) [inline stupidclient.js]: I didn't expect it to post as me, though

**Comment by riftEmber** [comment]
Just tested. Bot continues playing after I left the game

**Comment by riftEmber** [comment]
I played 3 games, 3 bots continued playing later. Also for some reason it is sending units from undefined nodes now:
2016-05-08-143103_1920x1080_scrot

**Comment by riftEmber** [comment]
Ok the bots can literally think. They started a new game and are now honing their skills. I'm not even kidding
2016-05-08-143301_1920x1080_scrot

**Comment by vpzomtrrfrt** [comment]
Oops, I think I know what I did wrong

**Comment by vpzomtrrfrt** [comment]
Okay, should actually quit now

File: stupidclient.js [hunk 3]
```diff
@@ -1,97 +1,123 @@
...
-	process.exit();
+		else if(ai && d.data == "gamestart") {
+			ai.active = true;
+		}
+	};
+	s.onclose = function() {
+		if(opts.log) {console.log("Lost connection");}
+		endGame();
+	};
+	var mint = setInterval(function(){
```
Comment (vpzomtrrfrt) [inline stupidclient.js:104]: Codacy Issue found: ["mint" is already defined](https://www.codacy.com/app/vpzomtrrfrt/turbo-spork-server/file/2532110848/issues/source?bid=3248399&fileBranchId=3284044#l104)

**Comment by riftEmber** [comment]
What are mint and kaint supposed to be, and why did it start a new game after I left previously?

**Comment by vpzomtrrfrt** [comment]
Those are the timers doing the AI stuff.  stupidclient normally tries to reconnect if it leaves so you can just leave it running and test stuff, but it shouldn't do that in this case.

**Pull request merged**